### PR TITLE
Document need to specify sharding for streaming mode

### LINKED ISSUE #245: Add comment to beam readme about when --outputNumShards is required

### PR DESCRIPTION
**Pull request by jklukas**
Closes #245

### CODE REVIEW

File: ingestion-beam/README.md [hunk 1]
```diff
@@ -303,6 +304,28 @@ gcloud dataflow jobs show "$JOB_ID"
...
+```
+
+When interacting with an unbounded source or sink like `pubsub`, Beam
+will implicitly set itself to streaming execution mode, which imposes some
+additional constraints on configuration of the underlying `FileIO.Write` transform.
+As discussed in the
+[Beam documentation for `FileIO.Write#withNumShards`](https://beam.apache.org/releases/javadoc/2.8.0/org/apache/beam/sdk/io/FileIO.Write.html#withNumShards-int-),
+batch mode is most efficient when the runner is left to determine sharding,
+so `numShards` options should normally be left to their default of `0`, but
+streaming mode can't perform the same optimizations and an exception will be thrown
```
Comment (whd) [inline ingestion-beam/README.md]: nit: thrown when?

**Comment by jklukas** [comment]
You make a good point about validating the condition in code. I'll look for the condition and raise a custom exception.

**Comment by jklukas** [comment]
Picked nits and added validation in the options parsing code.

File: ingestion-beam/src/main/java/com/mozilla/telemetry/options/SinkOptions.java [hunk 2]
```diff
@@ -147,6 +147,28 @@ static Parsed parseSinkOptions(SinkOptions options) {
...
+          + " --outputType=file and the input is unbounded (--inputType=pubsub);"
+          + " Dataflow recommends starting with twice the value of --maxWorkers or 10"
+          + " (see https://github.com/apache/beam/pull/1952)");
+    }
+    if (options.getInputType() == InputType.pubsub
+        && options.getErrorOutputType() == ErrorOutputType.file
+        && options.getErrorOutputNumShards() == 0) {
+      throw new IllegalArgumentException("Missing required parameter: "
+          + " --errorOutputNumShards must be set to an explicit non-zero value when"
+          + " --errorOutputType=file and the input is unbounded (--inputType=pubsub);"
```
Comment (whd) [inline ingestion-beam/src/main/java/com/mozilla/telemetry/options/SinkOptions.java]: There's a bunch of mostly duplicated string here but it's likely not worth the effort to dedupe it.

**Comment by jklukas** [comment]
> looks like it's only going to tell you you're missing the first one

Good usability point. I just squashed in a change to accumulate multiple validation errors and include them all in the message.

**Pull request merged**